menus/menu_pool.py

```diff
@@ -104,7 +103,9 @@ def __init__(self, pool, request):
         self.request = request
         self.request_language = get_language_from_request(request, check_path=True)
         self.site = Site.objects.get_current(request)
-        self.draft_mode_active = use_draft(request)
+        toolbar = getattr(request, "toolbar", None)
+        # No toolbar - no draft mode
+        self.draft_mode_active = toolbar.edit_mode_active or toolbar.preview_mode_active if toolbar else False
```

Comment:
The CMS has no concept of draft mode, this may be old logic that's never used (would have thrown an error if it did). Should be removed and if it is used anywhere replaced with bool(toolbar.edit_mode_active or toolbar.preview_mode_active) anywhere it is still referenced

. 
```suggestion
```

Reply:
This needs the toolbar which is not generally available in the menu. In practice the confusing thing is the legacy name `draft_mode_active`. In v4, the menu on the website is cached separately from the menu in edit or preview. I'd rather rename this.